server/src/pages/shot/view.js

```diff
@@ -452,6 +456,24 @@ class Body extends React.Component {
     </reactruntime.BodyTemplate>);
   }
 
+  promoOpen(promoType) {
+    if (promoType === "edit") {
+      document.querySelector(".button.transparent.edit").style.backgroundColor = "#ededf0";
```

Comment:
We definitely want to avoid altering other components directly like this. It makes it really hard to reason about the behavior of each component in isolation. Instead, whenever this promo card is shown, we want to also make sure the `highlightEditButton` state is set. The parent page is already managing both bits of state in `componentDidMount`:

```js
  componentDidMount() {
    this.setState({highlightEditButton: this.props.highlightEditButton});
    this.setState({promoDialog: this.props.promoDialog});
  }
```

All you need to do is ensure the highlight is set whenever the promo is visible. I think this should work:

```js
  componentDidMount() {
    this.setState({highlightEditButton: this.props.highlightEditButton || this.props.promoDialog});
    this.setState({promoDialog: this.props.promoDialog});
  }
```

Reply:
btw you can just apply the hover style to the edit button when `highlightEditButton` is true. That'd make the highlight a little more noticeable, even when the card isn't present.